src/codegen/compile.ml

```diff
@@ -179,25 +182,37 @@ module StaticBytes = struct
 
   type t_ =
     | I32 of int32
-    (* | I64 of int64 *)
+    | I64 of int64
     | Seq of t
     | Bytes of string
 
   and t = t_ list
 
-  let i32s is = Seq (List.map (fun i -> I32 i) is)
+  let i64s is = Seq (List.map (fun i -> I64 i) is)
 
   let rec add : Buffer.t -> t_ -> unit = fun buf -> function
     | I32 i -> Buffer.add_int32_le buf i
-    (* | I64 i -> Buffer.add_int64_le buf i *)
+    | I64 i -> Buffer.add_int64_le buf i
     | Seq xs -> List.iter (add buf) xs
     | Bytes b -> Buffer.add_string buf b
 
   let as_bytes : t -> string = fun xs ->
-    let buf = Buffer.create 16 in
+    let buf = Buffer.create 32 in
     List.iter (add buf) xs;
     Buffer.contents buf
 
+  let as_words static_bytes =
+    let rec convert_to_words binary index = 
+      assert (index <= (Bytes.length binary));
+      if (Bytes.length binary) = index then 
+        []
+      else 
+        let number = Bytes.get_int64_le binary index in
+        let next_index = Int.add index 8 in
+        [number] @ (convert_to_words binary next_index)
```

Comment:
(not tailrecursive, but only used for bignum constants so fine)

Reply:
Yes, if the constant bignum is not overly large, the stack recursion should be feasible.